SoftLayer/managers/block.py

```diff
@@ -0,0 +1,317 @@
+            ]
+            kwargs['mask'] = "mask[%s]" % ','.join(items)
+        return self.block_svc.getObject(id=volume_id, **kwargs)
+
+    def delete_snapshot(self, snapshot_id):
+        """Deletes the specified snapshot object.
+
+        :param snapshot_id: The ID of the snapshot object to delete.
+        """
+        return self.block_svc.deleteObject(id=snapshot_id)
+
+    def order_block_volume(self, storage_type, location, size, os_type,
+                           iops=None, tier_level=None):
+        """Places an order for a block volume.
+
+        :param storage_type: "Performance" or "Endurance"
+        :param location: Datacenter in which to order iSCSI volume
+        :param size: Size of the desired volume, in GB
+        :param os_type: OS Type to use for volume alignment, see help for list
+        :param iops: Number of IOPs for a "Performance" order
+        :param tier_level: Tier level to use for an "Endurance" order
+        """
+
+        base_type_name = 'SoftLayer_Container_Product_Order_Network_'
+        package = self._get_package(storage_type)
+        if package['name'] == 'Performance':
+            complex_type = base_type_name + 'PerformanceStorage_Iscsi'
+            prices = self._get_item_prices_performance(package, size, iops)
+        elif package['name'] == 'Endurance':
+            complex_type = base_type_name + 'Storage_Enterprise'
+            prices = self._get_item_prices_endurance(package, size, tier_level)
+        else:
+            raise exceptions.SoftLayerError(
+                "storage_type must be either Performance or Endurance")
+
+        location_id = self._get_location_id(location)
+
+        order = {
+            'complexType': complex_type,
+            'packageId': package['id'],
+            'osFormatType': {'keyName': os_type},
+            'prices': prices,
+            'quantity': 1,
+            'location': location_id,
+        }
+
+        return self.product_order.placeOrder(order)
+
+    def _get_package(self, category_code, **kwargs):
+        """Returns a product packaged based on type of storage.
+
+        :param category_code: Category code of product package.
+        :param kwargs:
+        :return: Returns a packaged based on type of storage.
+        """
+        if 'mask' not in kwargs:
+            items = [
+                'id',
+                'name',
+                'items',
+                'items[prices[categories],attributes]'
+            ]
+            kwargs['mask'] = "mask[%s]" % ','.join(items)
+
+        _filter = utils.NestedDict(kwargs.get('filter') or {})
+        _filter['categories']['categoryCode'] = (
+            utils.query_filter(category_code))
+        _filter['statusCode'] = (utils.query_filter('ACTIVE'))
+        kwargs['filter'] = _filter.to_dict()
+
+        func = getattr(self.product_package, 'getAllObjects')
+        return func(**kwargs).pop()
+
+    @staticmethod
```

Comment:
top-level functions are preferred over static methods. That reduces the amount of indentions by one which usually helps more than you'd expect. Thanks for using underscores to denote internal methods.